internal/cli/template_update.go

```diff
@@ -0,0 +1,233 @@
+package cli
+
+import (
+	"os"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateUpdateCommand struct {
+	*baseCommand
+
+	flagID   string
+	flagName string
+
+	flagSummary                    string
+	flagExpandedSummary            string
+	flagReadmeMarkdownTemplatePath string
+	flagWaypointHCLTemplatePath    string
+	flagTFCNoCodeModuleSource      string
+	flagTFCNoCodeModuleVersion     string
+	flagTags                       []string
+}
+
+func (c *ProjectTemplateUpdateCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	name := ""
+	if len(args) == 1 {
+		name = args[0]
+		return 1
+	}
+
+	var tref pb.Ref_ProjectTemplate
+	if name != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: name,
+		}
+	} else if c.flagID != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Id{
+			Id: c.flagID,
+		}
+		name = c.flagID
+	} else if c.flagName != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: c.flagName,
+		}
+		name = c.flagName
+	} else {
+		c.ui.Output("Missing project template name or id.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	checkResp, err := c.project.Client().GetProjectTemplate(ctx, &pb.GetProjectTemplateRequest{
+		ProjectTemplate: &pb.Ref_ProjectTemplate{
+			Ref: &pb.Ref_ProjectTemplate_Name{
+				Name: name,
+			},
+		},
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	if checkResp.ProjectTemplate == nil {
+		c.ui.Output(
+			"Project template %q does not exist", checkResp.ProjectTemplate.Name,
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
```

Comment:
Will this block ever trigger? From what I see if the template doesn't exist the service returns `codes.NotFound` error which would fall under the above error check. Should we move this check or specific error message under the error check above and include it if we hit `NotFound`?

```suggestion
	if err != nil {
	  errMsg := clierrors.Humanize(err)
	  if status.Code(err) == codes.NotFound || checkResp.ProjectTemplate == nil {
			errMsg = fmt.Sprintf("Project template %q does not exist", checkResp.ProjectTemplate.Name),
	  }
		c.ui.Output(errMsg, terminal.WithErrorStyle())
		return 1
	}
```

Reply:
A good catch, looks like we do indeed return a `NotFound` error here. Committing your suggested changes